src/SprykerEco/Zed/PunchoutCatalogs/Persistence/PunchoutCatalogsEntityManager.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Zed\PunchoutCatalogs\Persistence;
+
+use Generated\Shared\Transfer\PunchoutCatalogConnectionCartTransfer;
+use Generated\Shared\Transfer\PunchoutCatalogConnectionSetupTransfer;
+use Generated\Shared\Transfer\PunchoutCatalogConnectionTransfer;
+use Orm\Zed\PunchoutCatalog\Persistence\PgwPunchoutCatalogConnection;
+use Orm\Zed\PunchoutCatalog\Persistence\PgwPunchoutCatalogConnectionCart;
+use Orm\Zed\PunchoutCatalog\Persistence\PgwPunchoutCatalogConnectionSetup;
+use Spryker\Zed\Kernel\Persistence\AbstractEntityManager;
+
+/**
+ * @method \SprykerEco\Zed\PunchoutCatalogs\Persistence\PunchoutCatalogsPersistenceFactory getFactory()
+ */
+class PunchoutCatalogsEntityManager extends AbstractEntityManager implements PunchoutCatalogsEntityManagerInterface
```

Comment:
Please, check that you are not using lazy loading in mapper.
All checks like `$punchoutCatalogTransaction->getPunchoutCatalogConnection()` must be covered by `joinWith...` inrepository.

Reply:
Checked, they are already covered with `joinWith... `